src/types/textual.md

```diff
@@ -2,15 +2,20 @@
 
 The types `char` and `str` hold textual data.
 
-A value of type `char` is a [Unicode scalar value] (i.e. a code point that
-is not a surrogate), represented as a 32-bit unsigned word in the 0x0000 to
-0xD7FF or 0xE000 to 0x10FFFF range. A `[char]` is effectively a UCS-4 / UTF-32
-string.
+A value of type `char` is a [Unicode scalar value] (i.e. a code point that is
+not a surrogate), represented as a 32-bit unsigned word in the 0x0000 to 0xD7FF
+or 0xE000 to 0x10FFFF range. It is immediate [Undefined Behavior] to create a
+`char` that falls outside this range. A `[char]` is effectively a UCS-4 / UTF-32
+string of length 1.
 
 A value of type `str` is a Unicode string, represented as an array of 8-bit
-unsigned bytes holding a sequence of UTF-8 code points. Since `str` is a
-[dynamically sized type], it is not a _first-class_ type, but can only be
-instantiated through a pointer type, such as `&str`.
+unsigned bytes holding a sequence of UTF-8 code points. Note that this is a
```

Comment:
I'm uncertain, but is the correct term maybe "UTF-8 code units"?

Reply:
I don't know, I just kept the existing text here...

Reply:
But Wikipedia says you are right

> UTF-8 (8-bit Unicode Transformation Format) is a variable width character encoding capable of encoding all 1,112,064[nb 1] valid character code points in Unicode using one to four one-byte (8-bit) **code units**.

Reply:
Hm actually, "code points" might be correct here though -- that's the safety invariant. So I changed it ti "UTF-8 encoded Unicode code points" instead.